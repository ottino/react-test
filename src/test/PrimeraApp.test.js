import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';

const { render } = require("@testing-library/react");
const { default: PrimeraApp } = require("../PrimeraApp");

describe('Pruebas en componentes de react' , () => {

    // test('debe de mostrar el mensaje "Hola Crack"', () => {

    //     const saludo = 'Hola Crack';

    //     const wrapper = render( <PrimeraApp saludo={ saludo } />);


    // })


    test('debe de mostrar <PrimeraApp /> correctamente' , () => {

        const saludo = 'Hola Crack';
        const wrapper = shallow( <PrimeraApp saludo={ saludo }/> );

        expect( wrapper ).toMatchSnapshot();

    });


    test('debe de mostrar el subtitulo enviado por propos', () => {

        const saludo = 'Hola Crack';
        const subTitulo = 'Maxi';
        const wrapper = shallow( 
            <PrimeraApp 
                saludo ={ saludo }
                subtitulo = { subTitulo }
            /> 
        );

        const textoParrafo = wrapper.find('p').text();
        // console.log(textoParrafo);

        expect( textoParrafo ).toBe(subTitulo);


    });
})