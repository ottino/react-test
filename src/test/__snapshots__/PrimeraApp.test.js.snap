// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componentes de react debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola Crack
     
  </h1>
  <p>
    Maxi
  </p>
</Fragment>
`;
